src/System.CommandLine.Suggest/SuggestionStore.cs

```diff
@@ -44,7 +44,7 @@ public string GetCompletions(string exeFileName, string suggestionTargetArgument
 
                     Task<string> readToEndTask = process.StandardOutput.ReadToEndAsync();
 
-                    if (readToEndTask.Wait(timeout))
+                    if (readToEndTask.Wait(timeout) && process.HasExited && process.ExitCode == 0)
```

Comment:
The `else` block might need some additional checks as well. It doesn't appear that we have code coverage over both of these code branches.